File: server/router/live_chat/live_chat.go
package live_chat

import (
	"context"
	"sync"

	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/flipped-aurora/gin-vue-admin/server/router/live_chat/initialize"
	"github.com/gin-gonic/gin"
)

type LiveChatRouter struct{}

var once sync.Once

func (r *LiveChatRouter) InitLiveChatRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	once.Do(func() {
		ctx := context.Background()
		initialize.Menu(ctx)
		initialize.Api(ctx)
	})

	liveChatApi := v1.ApiGroupApp.LiveChatApiGroup

	// Write operations (with operation record)
	recordRouter := Router.Group("liveChat").Use(middleware.OperationRecord())
	{
		recordRouter.POST("product/create", liveChatApi.ProductApi.Create)
		recordRouter.PUT("product/update", liveChatApi.ProductApi.Update)
		recordRouter.DELETE("product/delete", liveChatApi.ProductApi.Delete)

		recordRouter.POST("faq/create", liveChatApi.FaqApi.Create)
		recordRouter.PUT("faq/update", liveChatApi.FaqApi.Update)
		recordRouter.DELETE("faq/delete", liveChatApi.FaqApi.Delete)
		recordRouter.POST("faq/import", liveChatApi.FaqApi.Import)

		recordRouter.POST("agent/online", liveChatApi.AgentApi.Online)
		recordRouter.POST("agent/offline", liveChatApi.AgentApi.Offline)
		recordRouter.PUT("agent/update", liveChatApi.AgentApi.Update)

		recordRouter.POST("chat/assign", liveChatApi.ChatApi.Assign)
		recordRouter.POST("chat/reply", liveChatApi.ChatApi.Reply)
		recordRouter.POST("chat/close", liveChatApi.ChatApi.Close)
	}

	// Read operations
	privateRouter := Router.Group("liveChat")
	{
		privateRouter.GET("product/list", liveChatApi.ProductApi.List)
		privateRouter.GET("product/:id", liveChatApi.ProductApi.Detail)

		privateRouter.GET("faq/list", liveChatApi.FaqApi.List)
		privateRouter.GET("faq/categories", liveChatApi.FaqApi.Categories)

		privateRouter.GET("agent/list", liveChatApi.AgentApi.List)
		privateRouter.GET("agent/status", liveChatApi.AgentApi.Status)

		privateRouter.GET("chat/sessions", liveChatApi.ChatApi.Sessions)
		privateRouter.GET("chat/session/:id", liveChatApi.ChatApi.SessionDetail)

		privateRouter.GET("report/overview", liveChatApi.ReportApi.Overview)
		privateRouter.GET("report/trend", liveChatApi.ReportApi.Trend)
	}
}

File: web/src/api/liveChat.js
import service from '@/utils/request'

// Product
export const createProduct = (data) => service({ url: '/liveChat/product/create', method: 'post', data })
export const updateProduct = (data) => service({ url: '/liveChat/product/update', method: 'put', data })
export const deleteProduct = (data) => service({ url: '/liveChat/product/delete', method: 'delete', data })
export const getProductList = (params) => service({ url: '/liveChat/product/list', method: 'get', params })
export const getProductDetail = (id) => service({ url: `/liveChat/product/${id}`, method: 'get' })

// FAQ
export const createFaq = (data) => service({ url: '/liveChat/faq/create', method: 'post', data })
export const updateFaq = (data) => service({ url: '/liveChat/faq/update', method: 'put', data })
export const deleteFaq = (data) => service({ url: '/liveChat/faq/delete', method: 'delete', data })
export const getFaqList = (params) => service({ url: '/liveChat/faq/list', method: 'get', params })
export const importFaq = (data) => service({ url: '/liveChat/faq/import', method: 'post', data })
export const getFaqCategories = (params) => service({ url: '/liveChat/faq/categories', method: 'get', params })

// Agent
export const agentOnline = (data) => service({ url: '/liveChat/agent/online', method: 'post', data })
export const agentOffline = (data) => service({ url: '/liveChat/agent/offline', method: 'post', data })
export const updateAgent = (data) => service({ url: '/liveChat/agent/update', method: 'put', data })
export const getAgentList = (params) => service({ url: '/liveChat/agent/list', method: 'get', params })
export const getAgentStatus = (params) => service({ url: '/liveChat/agent/status', method: 'get', params })

// Chat
export const getChatSessions = (params) => service({ url: '/liveChat/chat/sessions', method: 'get', params })
export const getSessionDetail = (id) => service({ url: `/liveChat/chat/session/${id}`, method: 'get' })
export const assignSession = (data) => service({ url: '/liveChat/chat/assign', method: 'post', data })
export const agentReply = (data) => service({ url: '/liveChat/chat/reply', method: 'post', data })
export const closeSession = (data) => service({ url: '/liveChat/chat/close', method: 'post', data })

// Report
export const getReportOverview = (params) => service({ url: '/liveChat/report/overview', method: 'get', params })
export const getReportTrend = (params) => service({ url: '/liveChat/report/trend', method: 'get', params })
